### PR TITLE
Add HTML formatter

### PR DESCRIPTION
**Pull request by mrb**
This is going to be very helpful for engine developers, primarily because it gives them a chance to see HTML rendered output versions of their description and content body.

screenshot 2016-05-20 18 02 19

### CODE REVIEW

File: lib/cc/analyzer/formatters/html_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            issues << json
+          when "warning"
+            warnings << json
+          else
+            raise "Invalid type found: #{json["type"]}"
+          end
+        end
+
+        def finished
+          puts "<h3>#{pluralize(issues.size, "issue")} found in #{issues_by_path.size} files.</h3>"
```
Comment (GordonDiggs) [inline lib/cc/analyzer/formatters/html_formatter.rb]: All these `puts` make me sad. How hard would it be to pull something like ERB into this to handle these?

Comment (mrb) [inline lib/cc/analyzer/formatters/html_formatter.rb]: @gordondiggs probably not too hard! I'll give it a shot.

**Comment by mrb** [comment]
@GordonDiggs Ready for re-review

File: lib/cc/analyzer/formatters.rb [hunk 2]
```diff
@@ -4,11 +4,13 @@ module Formatters
       autoload :Formatter, "cc/analyzer/formatters/formatter"
       autoload :JSONFormatter, "cc/analyzer/formatters/json_formatter"
       autoload :PlainTextFormatter, "cc/analyzer/formatters/plain_text_formatter"
+      autoload :HTMLFormatter, "cc/analyzer/formatters/html_formatter"
       autoload :Spinner, "cc/analyzer/formatters/spinner"
 
       FORMATTERS = {
         json: JSONFormatter,
         text: PlainTextFormatter,
+        html: HTMLFormatter,
```
Comment (wfleming) [inline lib/cc/analyzer/formatters.rb]: 💄 this list should probably be sorted, as should the autoload list above.

File: spec/cc/analyzer/formatters/html_formatter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+module CC::Analyzer::Formatters
+  describe HTMLFormatter do
+    include Factory
+
+    let(:formatter) do
+      filesystem ||= CC::Analyzer::Filesystem.new(
```
Comment (wfleming) [inline spec/cc/analyzer/formatters/html_formatter_spec.rb]: FWIW the return value of `let`s are memoized by rspec, so I don't think the `||` is doing anything for you here?

File: lib/cc/analyzer/formatters/templates/html.erb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  </head>
+  <body>
+  <div class="container">
+    <h2>Code Climate Analysis</h2>
+
+    <h3><%= pluralize(@issue_count, "issue") %> found in <%= @issues_by_path.size %> files.</h3>
+
+    <% @issues_by_path.each do |path, file_issues| %>
+      <hr>
+      <b><a name='<%= path %>'><%= path %></a> (<%= pluralize(file_issues.size, "issue") %>)</b>
```
Comment (GordonDiggs) [inline lib/cc/analyzer/formatters/templates/html.erb]: Maybe use an `h4` here?

**Comment by GordonDiggs** [comment]
One minor thing and I agree with Will's comment but otherwise LGTM! This will be awesome to have

**Comment by wfleming** [comment]
Definitely really helpful for community authors!

**Pull request merged**